### PR TITLE
fix: compaction support pick by max_seq

### PR DESCRIPTION
**Pull request by jiacai2050**
## Rationale
Part of #987.

Current implementation will compact by file size, max_seq is not considered, this may cause data corruption
in corner case, eg:
- sst1, max_seq:10, PK1=10
- sst2, max_seq:11, PK1=9
- sst3, max_seq:12, no PK1

If compact pick sst1 and sst3, and output sst4, its max_seq will be 12, now PK1 exists in two files:
- sst2, max_seq:11, PK1=9
- sst4, max_seq:12, PK1=10

That's to say, PK1's value is 10 now, which is wrong value(9 is right).

## Detailed Changes
When do compaction, first sort sst by max_seq desc, then only pick adjacent ssts, the original issue is fixed in this way.
At the same time picked ssts are ensured to meet other requirements such as `min_threshold`, `max_threshold`, `max_input_size`.

## Test Plan
UT and manually.

### CODE REVIEW

File: analytic_engine/src/compaction/picker.rs [hunk 1]
```diff
@@ -241,51 +262,87 @@ impl LevelPicker for SizeTieredPicker {
...
+                return files;
             }
+            debug!("No compaction necessary for segment, idx:{idx}, segment_key:{segment_key:?}");
         }
 
         None
     }
 }
 
 impl SizeTieredPicker {
```
Comment (Rachelint) [inline analytic_engine/src/compaction/picker.rs:280]: It's not STCS now, I think maybe we should define a trait having the `pick_ssts` method and place it into TWCS?

File: analytic_engine/src/compaction/picker.rs [hunk 2]
```diff
@@ -420,8 +477,17 @@ impl SizeTieredPicker {
 
 /// Time window compaction strategy
 /// See https://github.com/jeffjirsa/twcs/blob/master/src/main/java/com/jeffjirsa/cassandra/db/compaction/TimeWindowCompactionStrategy.java
-#[derive(Default)]
-pub struct TimeWindowPicker {}
+pub struct TimeWindowPicker {
+    pick_by_seq: bool,
+}
```
Comment (Rachelint) [inline analytic_engine/src/compaction/picker.rs:482]: Define the trait, and refactor like this?
```
struct TimeWindowPicker {
    new_bucket_picker: Box<dyn Picker>,
    old_bucket_picker: Box<dyn Picker>,
}
```

File: analytic_engine/src/compaction/picker.rs [hunk 3]
```diff
@@ -420,8 +477,17 @@ impl SizeTieredPicker {
 
 /// Time window compaction strategy
 /// See https://github.com/jeffjirsa/twcs/blob/master/src/main/java/com/jeffjirsa/cassandra/db/compaction/TimeWindowCompactionStrategy.java
```
Comment (Rachelint) [inline analytic_engine/src/compaction/picker.rs:479]: Modify comment?

File: analytic_engine/src/compaction/picker.rs [hunk 4]
```diff
@@ -176,10 +176,31 @@ fn trim_to_threshold(
...
+/// overlapping key range among them. Currently solution is relative simple,
+/// only pick adjacent sst. Maybe a better, but more complex solution could be
+/// introduced later.
+// TODO: Remove this function when pick_by_seq is stable.
+fn prefer_pick_by_seq() -> bool {
+    std::env::var("CERESDB_COMPACT_PICK_BY_SEQ").unwrap_or_else(|_| "true".to_string()) == "true"
+}
+
 /// Size tiered compaction strategy
 /// See https://github.com/jeffjirsa/twcs/blob/master/src/main/java/com/jeffjirsa/cassandra/db/compaction/SizeTieredCompactionStrategy.java
```
Comment (Rachelint) [inline analytic_engine/src/compaction/picker.rs:192]: Modify comment?

File: analytic_engine/src/compaction/picker.rs (hunk 2 above)
Comment (jiacai2050) [reply analytic_engine/src/compaction/picker.rs:482]: Not necessary, old implementation will be removed when new is stable.

File: analytic_engine/src/compaction/picker.rs (hunk 1 above)
Comment (jiacai2050) [reply analytic_engine/src/compaction/picker.rs:280]: Although it's not strict size tiered, but I would like to keep the structure as it's now, current implementation is relative simple, we may need to group sst by size when there are no overlapping. See comment at L182.

File: analytic_engine/src/compaction/picker.rs (hunk 3 above)
Comment (jiacai2050) [reply analytic_engine/src/compaction/picker.rs:479]: Already comment at L182 to explain current behavior, we can update this when new strategy is stable.

File: analytic_engine/src/compaction/picker.rs (hunk 1 above)
Comment (jiacai2050) [reply analytic_engine/src/compaction/picker.rs:280]: Moved here.

**Review by Rachelint — APPROVED** [review]
LGTM

**Pull request merged**